snuba/datasets/entities/discover.py

```diff
@@ -88,8 +81,32 @@ def attempt_map(
             return None
 
 
-@dataclass(frozen=True)
+@dataclass
 class DefaultNoneFunctionMapper(FunctionCallMapper):
+    """
+    Maps the list of function names to NULL.
+    """
+
+    function_names: Set[str]
+
+    def __post_init__(self) -> None:
+        self.function_match = FunctionCallMatch(
+            Or([StringMatch(func) for func in self.function_names])
+        )
+
+    def attempt_map(
+        self,
+        expression: FunctionCall,
+        children_translator: SnubaClickhouseStrictTranslator,
+    ) -> Optional[FunctionCall]:
+        if self.function_match.match(expression):
+            return identity(Literal(None, None), expression.alias)
+
+        return None
+
+
+@dataclass(frozen=True)
+class DefaultIfNullFunctionMapper(FunctionCallMapper):
```

Comment:
Hmm, I thought this whole function mapper was supposed to match against `ifNull` on a non existent column sent from the client - or is there some other purpose for it?

Reply:
I changed this back from `identity` to `ifNull` since there were more changes than I expected to make it work properly. We can tackle this at a later stage.